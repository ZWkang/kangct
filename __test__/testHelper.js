// import React from 'react';
import { mount, shallow, render } from 'enzyme';

function setUp(type, ...arg) {
  const [Component, props = {}, children = ''] = arg;
  if (~['mount', 'shallow', 'render'].indexOf(type)) {
    return global[type](<Component {...props}>{children}</Component>);
  }
}

export function AutoMath(Component, props = {}, children = '') {
  const wrapper = setUp('render', Component, props, children);
  expect(wrapper).toMatchSnapshot();
}

export function ComponentRenderWithoutError(Component, props = {}, children = '') {
  const wrapper = setUp('mount', Component, props, children);
  expect(() => {
    wrapper.setProps({});
    wrapper.unmount();
  }).not.toThrow();
}

export function childrenShouldBeProps(Component, children) {
  const wrapper = setUp('mount', Component, {}, children);

  expect(wrapper.get(0).props.children).toEqual(children);
}

export function setup(Component, props) {
  const wrapper = setUp(Component, props, null);
  return {
    wrapper,
    props
  };
}
