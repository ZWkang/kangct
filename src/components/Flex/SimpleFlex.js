import React from 'react'
import styled, { css } from 'styled-components'
import is from 'styled-is'

export default styled.div`
  display: flex;
`