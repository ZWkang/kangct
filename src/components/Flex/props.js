import PropTypes from 'prop-types'

export const FlexPropsTypes = {
  display: PropTypes.string,
  flexDirection: PropTypes.oneOf(['row', 'row-reverse', 'column', 'column-reverse']),
  flexWrap: PropTypes.oneOf(['nowrap', 'wrap', 'wrap-reverse']),
  flexFlow: PropTypes.string,
  justifyContent: PropTypes.oneOf(['flex-start', 'flex-end', 'center', 'space-between', 'space-around']),
  alignItems: PropTypes.oneOf(['flex-start', 'flex-end', 'center', 'baseline', 'stretch']),
  alignContent: PropTypes.oneOf(['flex-start', 'flex-end', 'center', 'space-between', 'space-around', 'stretch']),
  full: PropTypes.bool,
  center: PropTypes.bool,
  customStyle: PropTypes.string
}

export const FlexCommonPorps = {
  customStyle: ''
}


export const FlexItemPropsTypes = {
  order: PropTypes.number,
  flexGrow: PropTypes.number,
  flexShrink: PropTypes.number,
  flexBasis: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]),
  flex: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]),
  alignSelf: PropTypes.oneOf(['auto', 'flex-start', 'flex-end', 'center', 'baseline', 'stretch']),
  customStyle: PropTypes.string
}
// export default {
//   FlexPropsTypes,
//   FlexItemPropsTypes
// }