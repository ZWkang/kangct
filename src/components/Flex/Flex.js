import React from 'react';
import styled from 'styled-components';
import is from 'styled-is';
import { FlexPropsTypes, FlexCommonPorps } from './props.js';

// 样式分离
// css in js

const DefaultCssString = `
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: flex-start;
  align-content: stretch;
`;

const FullDefaultOption = `
  width: 100%;
  height: 100%;
  flex-basis: 100%;
`;

const CenterDefaultOption = `
  align-items: center;
  justify-content: center;
`;

const Flex = styled.div`
  ${DefaultCssString}

  ${is('display')`
    display: ${(props) => props.display}
  `}

  ${is('full')`
    ${FullDefaultOption}
  `}

  ${is('center')`
    ${CenterDefaultOption}
  `}

  ${is('flexDirection')`
    flex-direction: ${(props) => props.flexDirection}
  `}

  ${is('flexWrap')`
    flex-wrap: ${(props) => props.flexWrap}
  `}

  ${is('flexFlow')`
    flex-flow: ${(props) => props.flexFlow}
  `}

  ${is('justifyContent')`
    justify-content: ${(props) => props.justifyContent}
  `}

  ${is('alignItems')`
    align-items: ${(props) => props.alignItems}
  `}

  ${is('alignContent')`
    align-content: ${(props) => props.alignContent}
  `}

  ${is('customStyle')`
    ${(props) => props.customStyle}
  `}
`;

Flex.propTypes = FlexPropsTypes;
Flex.defaultProps = FlexCommonPorps;

/**
 * @components Flex
 */
export default Flex;
