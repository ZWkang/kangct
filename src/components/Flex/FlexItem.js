import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import is from 'styled-is'
import { FlexItemPropsTypes, FlexCommonPorps } from './props'

const FlexItemDefaultCssString = `
  order: 0;
  flex-basis: auto;
  flex-grow: 0;
  flex-shrink: 1;
  display: block;
`


const FlexItem = styled.div`

  ${FlexItemDefaultCssString}

  ${is('order')`
    order: ${props => props.order}
  `}

  ${is('flexGrow')`
    flex-grow: ${props => props.flexGrow}
  `}

  ${is('flexShrink')`
    flex-shrink: ${props => props.flexShrink}
  `}

  ${is('flexBasis')`
    flex-basis: ${props => props.flexBasis}
  `}

  ${is('flex')`
  flex: ${props => props.flex}
  `}


  ${is('alignSelf')`
    align-self: ${props => props.alignSelf}
  `}

  ${is('customStyle')`
    ${props => props.customStyle}
  `}
`

FlexItem.PropTypes = FlexItemPropsTypes
FlexItem.defaultProps = FlexCommonPorps

export default FlexItem