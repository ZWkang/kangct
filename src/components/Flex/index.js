import Flex from './Flex';
import FlexItem from './FlexItem';
import SimpleFlex from './SimpleFlex';

Flex.Item = FlexItem;
Flex.SimpleFlex = SimpleFlex;

export { FlexItem, Flex as default };
