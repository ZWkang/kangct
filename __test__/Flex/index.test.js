// import React from 'react';
// import { mount, shallow, render } from 'enzyme';
import Flex from '@app/Flex/index.js';
import { AutoMath, ComponentRenderWithoutError } from '../testHelper';
import 'jest-styled-components';

// https://github.com/airbnb/enzyme/issues/1798 dive vs shallow
describe('Flex', () => {
  describe('behavior test', () => {
    it('render without error', () => {
      ComponentRenderWithoutError(Flex);
    });
    it('test props snapshot', () => {
      AutoMath(
        () => (
          <Flex center>
            <Flex.Item order={3} flexBasis="20%">
              20%
            </Flex.Item>
            <Flex.Item order={2} flexBasis="40%">
              40%
            </Flex.Item>
            <Flex.Item order={1} flexBasis="40%">
              40%
            </Flex.Item>
          </Flex>
        ),
        {},
        null
      );
    });
  });
  describe('Flex props setting test', () => {
    let wrapper = null;
    beforeAll(() => {
      wrapper = mount(<Flex />, {});
    });
    it('align', () => {
      wrapper.setProps({
        center: true
      });

      expect(wrapper.find(Flex)).toHaveStyleRule('align-items', 'center');
      expect(wrapper.find(Flex)).toHaveStyleRule('justify-content', 'center');
    });
    it('full', () => {
      wrapper.setProps({
        full: true
      });

      expect(wrapper.find(Flex)).toHaveStyleRule('width', '100%');
      expect(wrapper.find(Flex)).toHaveStyleRule('height', '100%');
    });
    it('alignItems', () => {
      wrapper.setProps({
        alignItems: 'flex-start'
      });

      expect(wrapper.find(Flex)).toHaveStyleRule('align-items', 'flex-start');
    });
  });
});
