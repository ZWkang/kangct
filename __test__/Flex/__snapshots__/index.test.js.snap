// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex behavior test test props snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "sc-bdVaJa gBRtWL",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "sc-bwzfXH ghvxEw",
          "order": "3",
        },
        "children": Array [
          Object {
            "data": "20%",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "sc-bwzfXH gbyakH",
            "order": "2",
          },
          "children": Array [
            Object {
              "data": "40%",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": Object {
            "attribs": Object {
              "class": "sc-bwzfXH ipWma",
              "order": "1",
            },
            "children": Array [
              Object {
                "data": "40%",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": [Circular],
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "order": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "order": undefined,
            },
          },
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "order": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "order": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "order": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "order": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "sc-bwzfXH gbyakH",
          "order": "2",
        },
        "children": Array [
          Object {
            "data": "40%",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "sc-bwzfXH ipWma",
            "order": "1",
          },
          "children": Array [
            Object {
              "data": "40%",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "order": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "order": undefined,
          },
        },
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "sc-bwzfXH ghvxEw",
            "order": "3",
          },
          "children": Array [
            Object {
              "data": "20%",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "order": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "order": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "order": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "order": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "sc-bwzfXH ipWma",
          "order": "1",
        },
        "children": Array [
          Object {
            "data": "40%",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "sc-bwzfXH gbyakH",
            "order": "2",
          },
          "children": Array [
            Object {
              "data": "40%",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": Object {
            "attribs": Object {
              "class": "sc-bwzfXH ghvxEw",
              "order": "3",
            },
            "children": Array [
              Object {
                "data": "20%",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": [Circular],
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "order": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "order": undefined,
            },
          },
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "order": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "order": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "order": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "order": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
